internal/signer/darwin/keychain/keychain.go

```diff
@@ -405,3 +439,91 @@ func certIn(xc *x509.Certificate, xcs []*x509.Certificate) bool {
 	}
 	return false
 }
+
+func (k *Key) EncryptRSA(hashInput hash.Hash, random io.Reader, msg []byte) ([]byte, error) {
+	pub := k.Public()
+	rsaPubKey := pub.(*rsa.PublicKey)
+	return rsa.EncryptOAEP(hashInput, random, rsaPubKey, msg, nil)
+}
+
+func (k *Key) getRSAEncryptAlgor() (C.SecKeyAlgorithm, error) {
+	var algorithms map[crypto.Hash]C.CFStringRef
+	switch pub := k.Public().(type) {
+	case *rsa.PublicKey:
+		if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSASignatureDigestPSSSHA256) == 1 {
+			algorithms = rsaPSSAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSAEncryptionOAEPSHA256) == 1 {
+			algorithms = rsaOAEPAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeEncrypt, C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256) == 1 {
+			algorithms = rsaPKCS1v15Algorithms
+		}
+	default:
+		return 0, fmt.Errorf("unsupported algorithm %T", pub)
+	}
+	return algorithms[k.hashFunc], nil
+}
+
+func (k *Key) getEncryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	if k.hashFunc == 0 {
+		k.hashFunc = crypto.SHA256
+	}
+	return k.getRSAEncryptAlgor()
+}
+
+func (k *Key) getRSADecryptAlgor() (C.SecKeyAlgorithm, error) {
+	var algorithms map[crypto.Hash]C.CFStringRef
+	switch pub := k.Public().(type) {
+	case *rsa.PublicKey:
+		if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeDecrypt, C.kSecKeyAlgorithmRSASignatureDigestPSSSHA256) == 1 {
+			algorithms = rsaPSSAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeDecrypt, C.kSecKeyAlgorithmRSAEncryptionOAEPSHA256) == 1 {
+			algorithms = rsaOAEPAlgorithms
+		} else if C.SecKeyIsAlgorithmSupported(k.publicKeyRef, C.kSecKeyOperationTypeDecrypt, C.kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256) == 1 {
+			algorithms = rsaPKCS1v15Algorithms
+		}
+	default:
+		return 0, fmt.Errorf("unsupported algorithm %T", pub)
+	}
+	return algorithms[k.hashFunc], nil
+}
+
+func (k *Key) getDecryptAlgorithm() (C.SecKeyAlgorithm, error) {
+	return k.getRSADecryptAlgor()
+}
+
+func (k *Key) checkDataSize(plaintext []byte) error {
```

Comment:
I will check this today, we can commit it for now and come back to this.

Reply:
Just a FYI, I also made more progress on this (created the getPaddingSize())